### PR TITLE
#86 - Remove code duplication in IndexYaml

### LINKED ISSUE #86: IndexYaml.java:134-136: Remove duplicating code. There is...
The puzzle `84-6dc281f6` from #84 has to be resolved: 

https://github.com/artipie/helm-adapter/blob/3089ea716935645e06f6c9de47b9fca1571707aa/src/main/java/com/artipie/helm/metadata/IndexYaml.java#L134-L136

The puzzle was created by Aleksandr Krasnov on 11-Jan-21. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by genryxy**
Closes #86 
Removed code duplication in `IndexYaml`.
Added `IndexYamlMapping#addChart` to extract logic from `IndexYaml#update`

### CODE REVIEW

File: src/main/java/com/artipie/helm/metadata/IndexYaml.java [hunk 1]
```diff
@@ -216,4 +165,47 @@ private void update(final Map<String, Object> index, final TgzArchive tgz) {
...
+            options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+            options.setPrettyFlow(true);
+            return IndexYaml.this.storage.save(
+                IndexYaml.INDEX_YAML,
+                new Content.From(
+                    new Yaml(options).dump(index).getBytes(StandardCharsets.UTF_8)
+                )
+            );
+        }
+    }
```
Comment (olegmoz) [inline src/main/java/com/artipie/helm/metadata/IndexYaml.java:210]: @genryxy sorry, do not see a reason for this to be a class, it is not an entity. I believe it should be a private function

**Review by olegmoz — CHANGES_REQUESTED** [review]
@genryxy thanks, please see one design concern

**Comment by genryxy** [comment]
@olegmoz thanks, converted to function

**Review by olegmoz — APPROVED** [review]

**Comment by olegmoz** [comment]
@genryxy thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/helm-adapter#87` is not assigned, can't get performer

**Pull request merged**